src/pool/orca-pool.ts

```diff
@@ -162,18 +162,18 @@ export class OrcaPool {
       .compressIx(false);
     txBuilder.addInstruction(positionIx).addSigner(positionMintKeypair);
 
-    const { address: tokenOwnerAccountA, ...tokenOwnerAccountAIx } = await resolveOrCreateATA(
+    const [ataA, ataB] = await resolveOrCreateATAs(
+      this.dal,
       provider.connection,
       provider.wallet.publicKey,
-      whirlpool.tokenMintA,
-      maxTokenA
-    );
-    const { address: tokenOwnerAccountB, ...tokenOwnerAccountBIx } = await resolveOrCreateATA(
-      provider.connection,
-      provider.wallet.publicKey,
-      whirlpool.tokenMintB,
-      maxTokenB
+      [
+        { tokenMint: whirlpool.tokenMintA, wrappedSolAmountIn: maxTokenA },
+        { tokenMint: whirlpool.tokenMintB, wrappedSolAmountIn: maxTokenB },
```

Comment:
Although there is bug here, passing in wrappedSolAmountIn regardless of tokenMint feels a bit weird and dangerous to me. Maybe resolveOrCreateATAs should take wrappedSolAmountIn as an optional parameter and it can only be filled if tokenMint == native mint?